singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java

```diff
@@ -394,12 +394,18 @@ protected void injectHeadersForProducerRecord(LogMessage msg, Headers headers) {
   protected boolean checkMessageValid(LogMessage msg) {
     long singerChecksum = computeCRC(msg.getMessage());
     boolean isMessageValid = true;
-    // check if message is corrupted
-    if (msg.isSetChecksum() && singerChecksum != msg.getChecksum()) {
-      isMessageValid = false;
-      OpenTsdbMetricConverter.incr(SingerMetrics.NUM_CORRUPTED_MESSAGES, "topic=" + topic,
-              "host=" + HOSTNAME, "logName=" + msg.getLoggingAuditHeaders().getLogName(),
-              "logStreamName=" + logName);
+    if (msg.isSetChecksum()) {
+      // check if message is corrupted
+      if (singerChecksum != msg.getChecksum()) {
+        isMessageValid = false;
+        OpenTsdbMetricConverter.incr(SingerMetrics.NUM_CORRUPTED_MESSAGES, "topic=" + topic,
+                "host=" + HOSTNAME, "logName=" + msg.getLoggingAuditHeaders().getLogName(),
+                "logStreamName=" + logName);
+      } else {
+        OpenTsdbMetricConverter.incr(SingerMetrics.NUM_UNCORRUPTED_MESSAGES, "topic=" + topic,
+                "host=" + HOSTNAME, "logName=" + msg.getLoggingAuditHeaders().getLogName(),
+                "logStreamName=" + logName);
+      }
```

Comment:
This can be simplified I think; something like:
```
isMessageValid = singerChecksum == msg.getChecksum();
OpenTsdbMetricConverter.incr(isMessageValid ? SingerMetrics.NUM_UNCORRUPTED_MESSAGES : SingerMetrics.NUM_UNCORRUPTED_MESSAGES, "topic=" + topic,
                "host=" + HOSTNAME, "logName=" + msg.getLoggingAuditHeaders().getLogName(),
                "logStreamName=" + logName);
```